osv/tests.py

```diff
@@ -108,170 +108,108 @@ def commit(self, author_name, author_email, message='Changes'):
                              [self._repo.head.peel().id])
 
 
-_ds_data_dir = None
+class _DatastoreEmulator:
 
-_mock_gcs_ctx = None
+  def __init__(self, url):
+    self._url = url
+
+  def reset(self):
+    """Clear emulator data."""
+    resp = requests.post(f'http://{self._url}/reset', timeout=_EMULATOR_TIMEOUT)
+    resp.raise_for_status()
 
 
-def start_datastore_emulator():
-  """Starts Datastore emulator."""
-  # TODO(michaelkedar): turn this into a context (`with datastore_emulator()`)
-  _kill_existing_datastore_emulator()
+def _is_emulator_installed():
+  """Check if the gcloud firestore emulator is installed."""
+  try:
+    # We check if the component is installed.
+    output = subprocess.check_output([
+        'gcloud', 'components', 'list', '--quiet',
+        '--filter=id:cloud-firestore-emulator', '--format=value(state.name)'
+    ],
+                                     text=True,
+                                     stderr=subprocess.DEVNULL)
+    return 'Not Installed' not in output
+  except (subprocess.CalledProcessError, FileNotFoundError):
+    return False
+
+
+@contextlib.contextmanager
+def datastore_emulator():
```

Comment:
should we name this firestore emulator to avoid confusion? Or would that be confusing becuase it's firestore in datastore mode.

Reply:
eh I don't think so - it's still an emulator for datastore, just running on the firestore emulator (just like our database is 'Firestore with Datastore compatibility')